### PR TITLE
Add FAQ: update a field depending on another field

### LINKED ISSUE #269: Add FAQ section: How to update a field depending on another field
This is a question often asked. See here for an example https://jsonforms.discourse.group/t/creating-numeric-values-from-selected-criteria/1253

### PR DESCRIPTION
**Pull request by LukasBoll**
Add FAQ section: How to update a field depending on another field closes #269

### CODE REVIEW

File: content/faq/faq.mdx [hunk 1]
```diff
@@ -145,6 +147,107 @@ Now default values within the schema file can be used:
...
+      onChange={onChange}
+    />
+  );
+};
+```
+
+In the onChange method, the price is calculated based on the selected services. 
+If the calculated price matches the existing value, the data remains unchanged.
+However, if the price differs, a new data object is created with the updated price and passed to JSON Forms.
+It's important to only create a new object when necessary to avoid infinite loops.
```
Comment (sdirix) [inline content/faq/faq.mdx:280]: This sentence should be in a warning element. We already have some in the website.

File: content/faq/faq.mdx [hunk 2]
```diff
@@ -145,6 +147,107 @@ Now default values within the schema file can be used:
...
+The schema includes a multiselect field called services and a price attribute, with the price field set as readonly.
+
+<DependentFieldExample/>
+
+There are two approaches to update a field in JSON Forms based on the value of another field:
+
+#### Approach 1: Implementing functionality in the onChange method of JSON Forms
+
+In this approach, you can implement the logic in the onChange method of JSON Forms. 
+The onChange method triggers whenever a user changes the form data.
```
Comment (sdirix) [inline content/faq/faq.mdx:242]: Mention that this approach should only be used when using an older JSON Forms without middleware support as it works in a similar way, however the middleware variant is more performant.

**Review by sdirix — CHANGES_REQUESTED** [review]
Please add the middleware approach as approach number 1. In that approach mention that the similar approach 2 can be used for JSON Forms before version X without middleware support.

**Comment by sdirix** [comment]
Please resolve the conflicts by cherry-picking the lastest commit on `master`

File: content/faq/faq.mdx [hunk 3]
```diff
@@ -182,6 +184,160 @@ Now default values within the schema file can be used:
...
+```
+
+#### Approach 2: Implementing functionality in the onChange method of JSON Forms
+
+In this approach, you can implement the logic in the onChange method of JSON Forms. 
+The onChange method triggers whenever a user changes the form data.
+
+:::info Note
+
+Its more performant to use the middlewear approach, but this is a good approach for JSON Forms before version v3.2.0
```
Comment (sdirix) [inline content/faq/faq.mdx]:
```suggestion
It's more performant to use the middleware approach, but this is a good approach for JSON Forms before version v3.2.0
```

File: content/faq/faq.mdx [hunk 4]
```diff
@@ -182,6 +184,160 @@ Now default values within the schema file can be used:
...
+JSON Forms utilizes the reducer pattern and various actions to update its state.
+The middleware intercepts the call to the JSON Forms reducers and calls your custom code instead. 
+For detailed insights into the JSON Forms middleware, the reducer pattern, and JSON Forms actions, refer to the documentation [here](/docs/middleware). 
+In this scenario, we want to customize the behavior associated with the  `UPDATE_DATA` action, which is triggered when the form's data is changed by the user.
+We initially invoke JSON Forms default reducer to update the data and identify any errors.
+Subsequently, we adjust the price fields based on the selected services and update the state with the newly calculated data.
+We additionally override the `INIT` and `UPDATE_CORE` actions, in case the data prop passed to JSON Forms doesn't have the correct price set yet.
+
+```js
+import { INIT, UPDATE_DATA } from  '@jsonforms/core'
```
Comment (sdirix) [inline content/faq/faq.mdx]:
```suggestion
import { INIT, UPDATE_CORE, UPDATE_DATA } from  '@jsonforms/core'
```

File: content/faq/faq.mdx [hunk 5]
```diff
@@ -182,6 +184,160 @@ Now default values within the schema file can be used:
...
+
+```js
+export const CarWash = () => {
+  const [formData, setFormData] = useState(inputData);
+  
+  const onChange = ({ data, _errors }) => {
+    const price = data.services.length*15;
+    if (data.price === price){
+        setFormData(data);
+    }else{
```
Comment (sdirix) [inline content/faq/faq.mdx]:
```suggestion
    } else {
```

**Review by sdirix — APPROVED** [review]

**Comment by lucas-koehler** [comment]
@sdirix Can we merge this or is there a reason it is approved but not merged?

**Comment by sdirix** [comment]
We can merge, I just did a few minor changes and waited for the build to complete.

**Pull request merged**